app/components/osc-webhook-triggers/osc-webhook-triggers.html

```diff
@@ -1,38 +1,53 @@
+<ng-form name="$ctrl.secretsForm" class="add-webhook">
   <div ng-repeat="trigger in $ctrl.webhookTriggers">
     <div class="add-webhook-row">
-      <ui-select 
-        ng-model="trigger.data.type"
-        ng-disabled="$ctrl.isDeprecated(trigger)"
-        on-select="$ctrl.triggerTypeChange(trigger)"
-        search-enabled="false"
-        title="Select a webhook type"
-        class="select-webhook-type"
-        flex>
-        <ui-select-match placeholder="Webhook type">
-          {{ $select.selected.label }}
-        </ui-select-match>
-        <ui-select-choices repeat="option.label as option in $ctrl.webhookTypesOptions">
-          {{ option.label }}
-        </ui-select-choices>
-      </ui-select>
+      <div class="select-webhook-type">
+        <ui-select 
+          ng-model="trigger.data.type"
+          name="triggerType{{$index}}"
+          ng-disabled="$ctrl.isDeprecated(trigger)"
+          on-select="$ctrl.triggerTypeChange(trigger)"
+          search-enabled="false"
+          title="Select a webhook type"
+          ng-class="{'has-error': $ctrl.secretsForm.triggerType{{$index}}.$touched && !trigger.data.type && trigger.data[trigger.data.type.toLowerCase()].secretReference.name }"
+          focus-on="triggerTypeFocus{{$index}}"
+          flex>
+          <ui-select-match placeholder="Webhook type">
+            {{ $select.selected.label }}
+          </ui-select-match>
+          <ui-select-choices repeat="option.label as option in $ctrl.webhookTypesOptions">
+            {{ option.label }}
+          </ui-select-choices>
+        </ui-select>
+        <div class="has-error" ng-show="$ctrl.secretsForm.triggerType{{$index}}.$touched && !trigger.data.type && trigger.data[trigger.data.type.toLowerCase()].secretReference.name">
+          <span class="help-block">
+            Secret type is required.
+          </span>
+        </div>
+      </div>
 
-      <ui-select
-        ng-if="!$ctrl.isDeprecated(trigger)"
-        ng-model="trigger.data[trigger.data.type.toLowerCase()].secretReference.name"
-        ng-disabled="!trigger.data.type"
-        ng-required="trigger.data.type"
-        on-select="$ctrl.triggerSecretChange(trigger)"
-        title="Select a webhook secret reference"
-        class="select-secret-ref"
-        flex>
-        <ui-select-match placeholder="Webhook secret reference">
-          {{ $select.selected.metadata.name }}
-        </ui-select-match>
-        <ui-select-choices repeat="webhookSecret.metadata.name as webhookSecret in ($ctrl.webhookSecrets | filter : $select.search) track by (webhookSecret | uid)">
-          <div ng-bind-html="webhookSecret.metadata.name | highlight : $select.search"></div>
-        </ui-select-choices>
-      </ui-select>
+      <div class="select-secret-ref">
+        <ui-select
+          ng-if="!$ctrl.isDeprecated(trigger)"
+          ng-model="trigger.data[trigger.data.type.toLowerCase()].secretReference.name"
+          name="triggerSecretRef{{$index}}"
```

Comment:
This is probably a nit, but wondering if we would be helped by putting a prefix on our input names, like `inputTriggerSecretRef`.  I think for `keyValueEditor` we used utils that generated `names-with-hyphens` for the inputs, which helps a little bit as well.  Clues that help you understand you are dealing with form inputs rather than data variables are nice.

Reply:
Because of the [webhook trigger format](https://docs.openshift.org/latest/dev_guide/builds/triggering_builds.html#webhook-triggers):
```
type: "GitHub"
github:
  secretReference:
    name: "mysecret"
```

Reply:
Meant for the `name="triggerSecretRef{{$index}}"`, something like `name="trigger-secret-ref-input-{{index}}"`  This is more of a thought/suggestion, not sure its necessary.